src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1,4 +1,95 @@
+            +================+===========+=======+===================+
+            | 2017-01-01     | A         | 10    | [NAN]             |
+            +----------------+-----------+-------+-------------------+
+            | 2017-02-01     | A         | 4     | [10]              |
+            +----------------+-----------+-------+-------------------+
+            | 2017-03-01     | A         | 6     | [7]               |
+            +----------------+-----------+-------+-------------------+
+            | 2017-04-01     | A         | 11    | [5]               |
+            +----------------+-----------+-------+-------------------+
+
+            A more complex example would be, assuming we have maxHorizon = 2, window size = 2, min window periods = 2, and we want the mean
+            +----------------+-----------+-------+-------------------+
+            |time(YYYY-MM-DD)| grain     | target| target_max        |
+            +================+===========+=======+===================+
+            | 2017-01-01     | A         | 10    | [NAN, NAN]        |
+            +----------------+-----------+-------+-------------------+
+            | 2017-02-01     | A         | 4     | [NAN, NAN]        |
+            +----------------+-----------+-------+-------------------+
+            | 2017-03-01     | A         | 6     | [NAN, 7]          |
+            +----------------+-----------+-------+-------------------+
+            | 2017-04-01     | A         | 11    | [7, 5]            |
+            +----------------+-----------+-------+-------------------+
+
+    templates:
+      - name: T
+        types:
+            - int8
+            - int16
+            - int32
+            - int64
+            - uint8
+            - uint16
+            - uint32
+            - uint64
+            - float
+            - double
+
+    type_mappings:
+      - input_type: tuple<datetime, vector<string>, T>
+        output_type: vector<double>
+        is_output_optional: false
+        description: A vector of size maxHorizon will always be output, but the individual elements could be NaN.
+
+    custom_enums:
+      - name: AnalyticalRollingWindowCalculation
+        description: What calculation to use on the rolling window.
+        values: [Mean]
+        underlying_type: uint8
+
+    configuration_params:
+      - type: uint8
+        name: windowSize
+        is_optional: false
+        description: The maximum number of items in a window.
+
+      - type: uint8
+        name: minWindowCount
+        is_optional: false
+        description: The minimum number of elements that must be in an window to perform the calculation. If there is less, NaN will be returned.
+        
+      - type: AnalyticalRollingWindowCalculation
+        name: windowCalculation
+        is_optional: false
+        description: What calculation is going to be performed over the window data.
+
+      - type: uint32
+        name: maxHorizon

```

Comment:
Since we don't have horizon value in the execution() input data, do we still need the "max" prefix? There will be just one horizon all over the place.

Reply:
I can remove the max prefix. I am fine either way. Mentally max makes sense to me, but I think I am the outsider with that. I will remove it.